### PR TITLE
Add `markdown` module with `chat_to_html` and `media_to_html` 

### LINKED ISSUE #5177: [FEATURE] Migrate `media to html` utility methods to v2 sdk
**Is your feature request related to a problem? Please describe.**

To annotate media other than text, users need to represent the media as HTML. In the legacy SDK, this was supported by a set of utility methods, which can still be tunneled to via the new SDK:

```python
from argilla_v1.client.feedback.utils import audio_to_html, image_to_html, video_to_html, pdf_to_html
```

**Describe the solution you'd like**

This utility methods should be migrated and addressed within the v2 SDK.

### PR DESCRIPTION
**Pull request by davidberenstein1957**
# Pull Request Template

```python
from argilla.markdown import chat_to_html
```

- add a `markdown` module
- implements `chat_to_html` which formats openai chat messages to be rendered as HTML-like chat in Argilla
- re-implements `video/audio/image/pdf_to_html` methods from the v1 SDK.

Closes #5177 

**Type of change**

- New feature (non-breaking change which adds functionality)

**How Has This Been Tested**

N.A.

**Checklist**

- I added relevant documentation
- I followed the style guidelines of this project
- I did a self-review of my code
- I made corresponding changes to the documentation
- I confirm My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- I have added relevant notes to the CHANGELOG.md file (See https://keepachangelog.com/)

### CODE REVIEW

**Comment by davidberenstein1957** [comment]
Should be merged after https://github.com/argilla-io/argilla/pull/5189

**Review by sdiazlor — COMMENTED** [review]
LGTM!

File: argilla/tests/unit/test_markdown.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+import base64
+import unittest
+
+import pytest
+from argilla.markdown import audio_to_html, chat_to_html, image_to_html, pdf_to_html, video_to_html
+from argilla.markdown.media import _get_file_data, _is_valid_dimension, _media_to_html, _validate_media_type
+
+
+class TestMarkdown(unittest.TestCase):
```
Comment (frascuchon) [inline argilla/tests/unit/test_markdown.py]: We're using pytest for testing, and using the `assert` statement to validate tests. Maybe makes sense to align the test to the rest of the tests.

Comment (davidberenstein1957) [reply argilla/tests/unit/test_markdown.py]: @frascuchon, done. I got into the distilabel vibe.

File: argilla/src/argilla/markdown/media.py [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+    Raises:
+        FileNotFoundError: If the file does not exist or is empty.
+        ValueError: If no provided file type using bytes as input, the file type does not match the expected extension or the file size is too large (>5MB).
+    """
+    if isinstance(file_source, bytes):
+        if not file_type:
+            raise ValueError("File type must be provided if file source is a byte string.")
+        else:
+            file_type != "pdf" and _validate_media_type(media_type, file_type)
+        file_data = file_source
```
Comment (frascuchon) [inline argilla/src/argilla/markdown/media.py]:
```suggestion
        file_type != "pdf" and _validate_media_type(media_type, file_type)
        file_data = file_source
```

File: argilla/src/argilla/markdown/media.py [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+        autoplay: True to autoplay media. Defaults to False.
+        loop: True to loop media. Defaults to False.
+
+    Returns:
+        HTML tag with embedded base64 data.
+
+    Raises:
+        ValueError: If the width and height are not pixel or percentage.
+    """
+    if not _is_valid_dimension(width) or not _is_valid_dimension(height):
```
Comment (frascuchon) [inline argilla/src/argilla/markdown/media.py]:
```suggestion
    if not (_is_valid_dimension(width) or _is_valid_dimension(height)):
```

**Review by frascuchon — APPROVED** [review]
Just som code format suggestions.

**Pull request merged**